grails-app/services/au/org/emii/portal/BulkDownloadService.groovy

```diff
@@ -81,9 +91,8 @@ class BulkDownloadService {
 
             log.debug "Added $bytesCopied Bytes"
 
-            report.addSuccessfulFileEntry url, filenameToUse, bytesCopied
-        }
-        catch (Exception e) {
+            if (!isReportFile) { report.addSuccessfulFileEntry url, filenameToUse, bytesCopied }
+        } catch (Throwable e) {
```

Comment:
I'm not sure catching Throwable is justified here. Presumably this is to catch OutOfMemoryError but if that happens isn't the whole webapp going to be basically non-functional? I think we should allow the error to propagate to the top level asap rather than catching it here and treating it the same as if it had just been a faulty internet connection.

Reply:
No that catch is to catch failures when adding files to the zip so the correct entry can be generated for the report.  I changed it from Exception to Throwable because Throwable is the base class and hence will catch all problems.